samples/cellular/modem_shell/src/location/location_shell.c

```diff
@@ -672,15 +792,26 @@ static int cmd_location_cancel(const struct shell *shell, size_t argc, char **ar
 {
 	int ret;
 
-	gnss_location_to_cloud = false;
-	k_work_cancel(&gnss_location_work_data.work);
+	arg_cloud_details = false;
```

Comment:
Is it necessary to set these to `false`? Wouldn't it be just enough to cancel the work item?

And I'm also wondering about storing the "cancel" command below. Why not keep the current data and just try to cancel the work item? If the work item is already being executed, let it send the existing data.

Reply:
> Is it necessary to set these to `false`? Wouldn't it be just enough to cancel the work item?

No and yes. And we actually decided to remove cancelling the work item and let it be sent if scheduled.

> And I'm also wondering about storing the "cancel" command below. Why not keep the current data and just try to cancel the work item? If the work item is already being executed, let it send the existing data.

I've been wondering about storing "cancel command too 😄
I'm removing it. Need to discuss this still with @jhirsi.